### PR TITLE
Improvement BasePaymentPage

### PR DESCRIPTION
**Pull request by jechandia**
### Details

The Add Payment Methods button will align to the bottom edge and menu popups will align to the right

### Fixed Issues

https://github.com/Expensify/App/issues/8929

### Tests

- [x] Verify that the Add Payment Methods button aligns to the bottom edge
- [x] Check that the menus are positioned correctly when resizing the browser
- [x] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### QA Steps

1. Open app
2. Log in any account
3. Setting>Payments
4. Verify that the Add Payment Methods button aligns to the bottom edge
5. Show any payment menu
6. If it is on desktop/web, modify the size of the browser/app and verify that the popup menus are positioned correctly
7. Check that the menus are positioned correctly when resizing the browser

#### Web

Safari

https://user-images.githubusercontent.com/16966456/190695900-fc67bfb1-a4f0-4521-a6ed-d5100fccc18a.mov

Chrome

https://user-images.githubusercontent.com/16966456/190696767-4ff16304-c396-494f-9255-fd287acb7f1f.mov

Mozila

https://user-images.githubusercontent.com/16966456/190697110-43f8589b-fc50-4356-aa99-149c0d659615.mov

#### Mobile Web

https://user-images.githubusercontent.com/16966456/190697729-c5922f61-aed7-4cfa-b104-c11508f3d394.mp4

#### Desktop

https://user-images.githubusercontent.com/16966456/190698644-b8866f89-f559-49be-a624-4fcbe4f97e10.mov

#### iOS

https://user-images.githubusercontent.com/16966456/190699574-9c1ad019-ea88-4953-91e2-2380953e2af3.mp4

#### Android

https://user-images.githubusercontent.com/16966456/190700329-6266114d-40c8-429c-8f22-781f82763a49.mov

### CODE REVIEW

File: src/pages/settings/Payments/PaymentMethodList.js [hunk 1]
```diff
@@ -151,23 +151,6 @@ class PaymentMethodList extends Component {
             });
         }
 
-        if (!this.props.shouldShowAddPaymentMethodButton) {
-            return combinedPaymentMethods;
-        }
-
-        combinedPaymentMethods.push({
```
Comment (mananjadhav) [inline src/pages/settings/Payments/PaymentMethodList.js:158]: Because we're removing this, we would also have to remove the check from `renderItem`, `if (item.type === BUTTON) {`

File: src/pages/settings/Payments/PaymentMethodList.js [hunk 2]
```diff
@@ -235,11 +218,29 @@ class PaymentMethodList extends Component {
...
+                    this.props.shouldShowAddPaymentMethodButton
+                    && (
+                        <Button
+                            text={this.props.translate('paymentMethodList.addPaymentMethod')}
+                            icon={Expensicons.CreditCard}
+                            onPress={e => this.props.onPress(e)}
+                            isDisabled={this.props.isLoadingPayments}
+                            style={[styles.mb4, styles.mh4]}
+                            success
+                            shouldShowRightIco
```
Comment (mananjadhav) [inline src/pages/settings/Payments/PaymentMethodList.js:238]: There's a typo here, `shouldshowRightIcon` and hence `>` isn't showing up in the button.

File: src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js [hunk 3]
```diff
@@ -45,7 +45,8 @@ class BasePaymentsPage extends React.Component {
             selectedPaymentMethod: {},
             formattedSelectedPaymentMethod: {},
             anchorPositionTop: 0,
-            anchorPositionLeft: 0,
+            anchorPositionBottom: 0,
+            anchorPositionRigth: 0,
```
Comment (mananjadhav) [inline src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js:49]:
```suggestion
            anchorPositionRight: 0,
```

Can you also correct the typo at other places?

File: src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js [hunk 4]
```diff
@@ -109,10 +110,24 @@ class BasePaymentsPage extends React.Component {
      */
     setPositionAddPaymentMenu(position) {
         this.setState({
-            anchorPositionTop: position.bottom,
+            anchorPositionTop: position.top + position.height,
 
             // We want the position to be 13px to the right of the left border
-            anchorPositionLeft: position.left + 13,
+            anchorPositionRigth: (window.innerWidth - position.right) + 13,
```
Comment (mananjadhav) [inline src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js:116]: Can we confirm why this is supposed to be `window.innerWidth`

**Review by mananjadhav — CHANGES_REQUESTED** [review]
Requested small changes, rest on web looks fine. Once fixed, I'll test on other platforms.

**Comment by JosueEchandiaAsto** [comment]
Small requested changes made

**Comment by mananjadhav** [comment]
I'll check this today and test.

File: src/pages/settings/Payments/PaymentMethodList.js [hunk 5]
```diff
@@ -235,11 +202,29 @@ class PaymentMethodList extends Component {
...
     render() {
         return (
-            <FlatList
-                data={this.createPaymentMethodList()}
-                renderItem={this.renderItem}
-                keyExtractor={item => item.key}
-            />
+            <>
+                <FlatList
+                    bounces={false}
```
Comment (parasharrajat) [inline src/pages/settings/Payments/PaymentMethodList.js:207]: Any reason?

File: src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js [hunk 6]
```diff
@@ -109,10 +110,24 @@ class BasePaymentsPage extends React.Component {
...
+    }
+
+    /**
+     * Set position of the payment method menu
+     *
+     * @param {Object} position
+     */
+    setPositionAddPaymentMethod(position) {
+        this.setState({
+            anchorPositionBottom: window.innerHeight - position.top,
```
Comment (parasharrajat) [inline src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js:127]: :scream:

File: src/pages/settings/Payments/PaymentMethodList.js (hunk 5 above)
Comment (JosueEchandiaAsto) [reply src/pages/settings/Payments/PaymentMethodList.js:207]: The button was requested to be outside of the FlatList https://github.com/Expensify/App/issues/8929#issuecomment-1154362227

File: src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js (hunk 6 above)
Comment (JosueEchandiaAsto) [reply src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js:127]: Any change you suggest? should I avoid window.innerHeight and look for some alternative?

File: src/pages/settings/Payments/PaymentMethodList.js (hunk 5 above)
Comment (parasharrajat) [reply src/pages/settings/Payments/PaymentMethodList.js:207]: I meant to ask why are we disabling bounce.

File: src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js (hunk 6 above)
Comment (parasharrajat) [reply src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js:127]: You can use withWindowDimensions HOC to get these.

File: src/pages/settings/Payments/PaymentMethodList.js (hunk 5 above)
Comment (JosueEchandiaAsto) [reply src/pages/settings/Payments/PaymentMethodList.js:207]: To avoid the problem that was initially https://github.com/Expensify/App/issues/8929#issue-1231739326
_The button "Add payment method" is not visible when scrolled up_

File: src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js (hunk 6 above)
Comment (JosueEchandiaAsto) [reply src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js:127]: Oh sure, thank you very much. I'll make the change

File: src/pages/settings/Payments/PaymentMethodList.js (hunk 5 above)
Comment (JosueEchandiaAsto) [reply src/pages/settings/Payments/PaymentMethodList.js:207]: uhm of course, then I remove `bounces={false}`

**Comment by JosueEchandiaAsto** [comment]
@mananjadhav @parasharrajat 
Can I delete dimensionsSubscription? It is intersecting with the withWindowDimensions listener. It seems unnecessary to me because withWindowDimensions will be used.
https://github.com/Expensify/App/blob/75e4731b191c8d9e59d19bb1888642dfeb97f584/src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js#L67

**Comment by roryabraham** [comment]
> Can I delete dimensionsSubscription? It is crossing with the withWindowDimensions listener. It seems unnecessary to me because withWindowDimensions will be used.

Yeah, you can replace that with the `withWindowDimensions` listener and just listen for prop changes in the window dimensions in `componentDidUpdate` 👍

File: src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js (hunk 6 above)
Comment (JosueEchandiaAsto) [reply src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js]: Change made

File: src/pages/settings/Payments/PaymentMethodList.js (hunk 5 above)
Comment (JosueEchandiaAsto) [reply src/pages/settings/Payments/PaymentMethodList.js]: change made

**Comment by JosueEchandiaAsto** [comment]
the following changes were made
- Avoiding debounce on `PaymentMethodList`
- `dimensionsSubscription` replaced with the withWindowDimensions listener
- Avoided creating the `setPositionAddPaymentMenu` method but continuing to use `setPositionAddPaymentMethod` that was used initially

**Comment by roryabraham** [comment]
@JosueEchandiaAsto Just FYI you've got conflicts in this PR

**Comment by JosueEchandiaAsto** [comment]
@roryabraham 
Ok, I'll check it out. Can i create another PR?

**Comment by JosueEchandiaAsto** [comment]
Resolved conflicts. Any additional observation?

**Comment by JosueEchandiaAsto** [comment]
@mananjadhav wait a moment, in the main branch there is a change that uses FormAlertWrapper, I'm doing the update 
https://github.com/Expensify/App/blob/ff1b9cddd31e45226437b4e749b7cd64f6d33918/src/pages/settings/Payments/PaymentMethodList.js#L214-L228

**Comment by mananjadhav** [comment]
Thanks for picking it up @JosueEchandiaAsto

**Comment by JosueEchandiaAsto** [comment]
All ready

**Review by mananjadhav — DISMISSED** [review]
LGTM, tests well.  Thank you for your time and patience on this one @JosueEchandiaAsto.

@roryabraham I would like to highlight two points (changes unrelated to the PR):

1. I can see two console errors on this page, but they exist in the `main` branch as well and are unrelated to this PR
console-error-1

console-error-2


2. The size of the `Add Payment method` button is also smaller with the latest `main` changes compared to production.

button-size


🎀 👀 🎀 C+ reviewed

<details>
<summary><h4>PR Reviewer Checklist</h4>

The Contributor+ will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/docs/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/docs/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/docs/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/docs/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

**Comment by mananjadhav** [comment]
@JosueEchandiaAsto There are some merge conflicts here. Can you please take the latest changes and resolve them?

@roryabraham Quick bump on your review.

**Comment by JosueEchandiaAsto** [comment]
@mananjadhav new conflicts resolved

**Comment by parasharrajat** [comment]
Bump @roryabraham, it seems like it is waiting for your review.

**Comment by roryabraham** [comment]
Whoops, my apologies everyone I'm not sure how I missed this. Unfortunately it's got new conflicts but I'll review + test as-is so we can hopefully move quicker once the conflicts are resolved. My apologies again

**Comment by JosueEchandiaAsto** [comment]
@roryabraham @mananjadhav @parasharrajat new conflicts resolved

**Review by roryabraham — CHANGES_REQUESTED** [review]
After this change, we only have one type of item that goes in this list: `MENU_ITEM`. The only other non-menuItem  is added [right here](https://github.com/Expensify/App/pull/9622/files#diff-eec5ba837852ed6fbe3e2ee4fe371c45b643c59932f6abe38aaf665329a74219L140-L145) on lines 140-145, when the list rendered by the `FlatList` is empty.

`FlatList` has a prop just for this called `ListEmptyComponent`, so we should use that.

In summary, the changes I'm requesting are:

- Get rid of the `type` field on items in the `PaymentMethodList` and the `MENU_ITEM` const.
- Use the `ListEmptyComponent` to render the `addFirstPaymentMethod` test if the list is empty
- Get rid of the conditional rendering in `renderItem`

Thanks!

**Comment by mananjadhav** [comment]
@JosueEchandiaAsto Did you get a chance to look at @roryabraham's comment?

**Comment by JosueEchandiaAsto** [comment]
@mananjadhav Yes, I will make the change, along with the new conflicts again.

**Comment by mananjadhav** [comment]
@JosueEchandiaAsto Do we have any updates on the PR?

**Comment by JosueEchandiaAsto** [comment]
@roryabraham Requested changes are done

**Review by mananjadhav — CHANGES_REQUESTED** [review]
Few small comments @JosueEchandiaAsto. Apart from that can you please:

1. Take the latest `main` pull
2. Screencasts seem to be outdated (missing > in the button text). Can you please update the screenshot/videos with the latest rerun.

**Comment by JosueEchandiaAsto** [comment]
@mananjadhav Requested changes are done. Screencasts updated

**Comment by mananjadhav** [comment]
Thanks @JosueEchandiaAsto for the changes.

@roryabraham The code looks good to me and tests well.

🎀 👀 🎀 
C+ reviewed

<details>
<summary><h4>PR Reviewer Checklist</h4>

The Contributor+ will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

**Comment by mananjadhav** [comment]
@roryabraham Quick bump.

**Review by roryabraham — APPROVED** [review]

**Review by roryabraham — APPROVED** [review]
- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

**Comment by roryabraham** [comment]
Contributor had used an out-of-date checklist, and I chose to merge anyways for the sake of velocity.
